gslb/test/integration/custom_fqdn/lib.go

```diff
@@ -654,7 +667,54 @@ func getDefaultHostRule(name, ns, lfqdn, gfqdn, status string) *akov1alpha1.Host
 	}
 }
 
-func deleteHostRule(t *testing.T, cluster int, ns, name string) {
+func getHostRuleForCustomFqdn(name, ns, lfqdn, gfqdn, status string) *akov1alpha1.HostRule {
+	hr := getDefaultHostRule(name, ns, lfqdn, status)
+	hr.Spec.VirtualHost.Gslb = akov1alpha1.HostRuleGSLB{
+		Fqdn: gfqdn,
+	}
+	return hr
+}
+
+func getDefaultAliases(objType string) []string {
+	return []string{
+		objType + "_alias1" + ".avi.com",
+		objType + "_alias2" + ".avi.com",
+		objType + "_alias3" + ".avi.com",
+	}
+}
+
+func getHostRuleWithAliases(name, objType, ns, lfqdn, status string, aliases []string) *akov1alpha1.HostRule {
+	if aliases == nil {
+		aliases = getDefaultAliases(objType)
+	}
+	hr := getDefaultHostRule(name, ns, lfqdn, status)
+	hr.Spec.VirtualHost.Aliases = aliases
+	return hr
+}
+
+func getHostRuleWithAliasesForCustomFqdn(name, objType, ns, lfqdn, gfqdn, status string, aliases []string, includeAliases bool) *akov1alpha1.HostRule {
+	if aliases == nil {
+		aliases = getDefaultAliases(objType)
+	}
+	hr := getHostRuleForCustomFqdn(name, ns, lfqdn, gfqdn, status)
+	hr.Spec.VirtualHost.Gslb.IncludeAliases = includeAliases
+	hr.Spec.VirtualHost.Aliases = aliases
+	return hr
+}
+
+func getDefaultExpectedDomainNames(gsName string, hrObjList []*akov1alpha1.HostRule) []string {
+	aliasList := []string{}
+	for _, hr := range hrObjList {
+		aliasList = append(aliasList, hr.Spec.VirtualHost.Aliases...)
+	}
+	aliasSet := sets.NewString()
+	for _, alias := range aliasList {
+		aliasSet.Insert(alias)
+	}
+	return aliasSet.Insert(gsName).List()
```

Comment:
`hr.Spec.VirtualHost.Aliases` can be directly inserted into the set. `sets.NewString(hr.Spec.VirtualHost.Aliases...)` can be used here.